packages/rum-core/src/domain/rumEventsCollection/resource/resourceCollection.ts

```diff
@@ -76,12 +84,18 @@ function processResourceEntry(entry: RumPerformanceResourceTiming) {
         type,
         url: entry.name,
       },
-      type: RumEventType.RESOURCE,
+      type: RumEventType.RESOURCE as const,
     },
     tracingInfo,
     entryMetrics
   )
-  return { startTime: startClocks.relative, rawRumEvent: resourceEvent as RawRumResourceEvent }
+  return {
+    startTime: startClocks.relative,
+    rawRumEvent: resourceEvent,
+    domainContext: {
+      performanceEntry: (entry as unknown) as PerformanceEntry,
```

Comment:
This is not a good cast, since for `initial_document`, `entry` is not a PerformanceEntry